src/components/Legend/Legend.tsx

```diff
@@ -179,203 +103,113 @@ const StyledCardDescriptionSecond = styled(CardDescription)`
-            <PumpLabel>{'verriegelt'}</PumpLabel>
-          </FlexRow>
-
-          <FlexRow>
-            <PumpsDot color={defaultColor.hex} size={13} />
-            <PumpLabel>{'unbekannt'}</PumpLabel>
-          </FlexRow>
+          {visibleLayer === 'trees' && <TreesColorLegend />}
+          {visibleLayer === 'rain' && <RainColorLegend />}
         </UnstyledFlex>
       )}
-      {legendExpanded && (
-        <FlexColumn isLast={true}>
-          <StyledCardDescription>Datenpunkte</StyledCardDescription>
-          <StyledCardDescriptionSecond>
-            durch Klick ein- & ausblenden.
-          </StyledCardDescriptionSecond>
-          <UnstyledFlexWidth
-            active={treesVisible}
-            onClick={() => {
-              store.setState({ treesVisible: !treesVisible });
-              store.setState({ pumpsVisible: false });
-              store.setState({ rainVisible: false });
-            }}
-          >
-            {treesVisible === true ? (
-              <LegendDot
-                className={'legend-dot'}
-                color={'#2c303b'}
-                gradient={rainGradient}
-              />
-            ) : (
-              <LegendDot className={'legend-dot'} color={'#2c303b'} />
-            )}
-            <StyledItemLabel>Straßen- & Anlagenbäume</StyledItemLabel>
-          </UnstyledFlexWidth>
-          <UnstyledFlexWidth
-            active={pumpsVisible}
-            onClick={() => {
-              store.setState({ treesVisible: false });
-              store.setState({ pumpsVisible: !pumpsVisible });
-              store.setState({ rainVisible: false });
-            }}
-          >
-            {pumpsVisible === true ? (
-              <StrokedLegendDot gradient={`${workingColor.hex}`} />
-            ) : (
-              <StrokedLegendDot />
-            )}
-            <StyledItemLabel>Öffentl. Pumpen</StyledItemLabel>
-          </UnstyledFlexWidth>
-          <UnstyledFlexWidth
-            active={rainVisible}
-            onClick={() => {
-              store.setState({ treesVisible: false });
-              store.setState({ pumpsVisible: false });
-              store.setState({ rainVisible: !rainVisible });
-            }}
-          >
-            {rainVisible === true ? (
-              <LegendRect gradient={rainGradient} />
-            ) : (
-              <LegendRect />
-            )}
-            <StyledItemLabel>Niederschlagsflächen</StyledItemLabel>
-          </UnstyledFlexWidth>
-        </FlexColumn>
-      )}
+      {visibleLayer === 'pumps' && <PumpsColorLegend />}
+
+      <FlexColumn isLast={true}>
+        <StyledCardDescription>Datenpunkte</StyledCardDescription>
+        <StyledCardDescriptionSecond>
+          durch Klick ein- & ausblenden.
+        </StyledCardDescriptionSecond>
+        <UnstyledFlexWidth
+          active={visibleLayer === 'trees' ? true : false}
+          onClick={() => {
+            store.setState({ visibleLayer: 'trees' });
```

Comment:
There should be an action to set the visible layer. Please don't use `store.setState`.